packages/directlinespeech/src/createAdapters.js

```diff
@@ -155,6 +166,31 @@ export default async function create({
     }, TOKEN_RENEWAL_INTERVAL);
   }
 
+  // Renew token per interval.
+  if (enableInternalHttpSupport) {
+    const interval = setInterval(async () => {
+      // #2660 If the connector has been disposed, we should stop renewing the token.
+
+      // TODO: We should use a public implementation if Speech SDK has one related to "privIsDisposed".
+      if (dialogServiceConnector.privIsDisposed) {
+        clearInterval(interval);
+      }
+
+      const refreshedDirectLineToken = await refreshDirectLineToken(directLineToken);
+
+      if (!refreshedDirectLineToken) {
+        return console.warn(
+          'botframework-directlinespeech-sdk: Renew token failed because call to refresh token Direct Line API did not return a new token.'
+        );
+      }
+
+      config.setProperty(PropertyId.Conversation_ApplicationId, refreshedDirectLineToken);
+      dialogServiceConnector = patchDialogServiceConnectorInline(new DialogServiceConnector(config, audioConfig));  // eslint-disable-line require-atomic-updates
+      dialogServiceConnector.connect(); // eslint-disable-line require-atomic-updates
```

Comment:
Don't disable ESLint because its complaint is legit. It's partly catching the re-assignment issue I mentioned above.